packages/gatsby/src/utils/handle-flags.ts

```diff
@@ -16,11 +18,54 @@ const handleFlags = (
   flags: Array<IFlag>,
   configFlags: Record<string, boolean>,
   executingCommand = process.env.gatsby_executing_command
-): { enabledConfigFlags: Array<IFlag>; message: string } => {
+): {
+  enabledConfigFlags: Array<IFlag>
+  unknownFlagMessage: string
+  message: string
+} => {
   // Prepare config flags.
   // Filter out any flags that are set to false.
   const availableFlags = new Map()
   flags.forEach(flag => availableFlags.set(flag.name, flag))
+
+  // Find unknown flags someone has in their config to warn them about.
+  const unknownConfigFlags = Object.keys(configFlags)
+    .filter(flagName => !availableFlags.has(flagName))
+    .map(flag => {
+      const flagsWithDistance = flags.map(f => {
+        return {
+          name: f.name,
+          distance: distance(flag, f.name),
+        }
+      })
+
+      const minDistance = _.minBy(flagsWithDistance, f => f.distance)
+
+      let didYouMean
+      if (minDistance) {
+        didYouMean = minDistance.distance < 4 ? minDistance.name : undefined
+      }
+
+      return {
+        flag,
+        didYouMean,
+      }
+    })
+
+  let unknownFlagMessage = ``
+  if (unknownConfigFlags.length > 0) {
+    unknownFlagMessage = commaListsAnd`The following flag(s) found in your gatsby-config.js are not known: ${unknownConfigFlags.map(
+      f => f.flag
+    )}`
+
+    const didYouMeans = unknownConfigFlags.filter(f => f.didYouMean)
+    if (didYouMeans.length > 0) {
+      unknownFlagMessage += `\n\n${commaListsAnd`Did you mean: ${didYouMeans.map(
+        f => f.didYouMean
+      )}`}?\n`
+    }
+  }
```

Comment:
(from test snapshot above)
```
The following flag(s) found in your gatsby-config.js are not known: FASTLY_DEV and SUPER_COOL_FLAG

Did you mean: FAST_DEV?
```

wonder if it just wouldn't be more readable if it was outputting something like:
```
The following flag(s) found in your gatsby-config.js are not known: FASTLY_DEV (did you mean: FAST_DEV?) and SUPER_COOL_FLAG
```

or more "verbose":
```
The following flag(s) found in your gatsby-config.js are not known: 
 - FASTLY_DEV (did you mean: FAST_DEV?)
 - SUPER_COOL_FLAG
```

Reply:
ah nice, will change